### ISSUE TITLE
Enable ICU provider

### ISSUE THREAD

**Issue by lukaspili** [supabase/postgres#198]
One popular use case is to create a case insensitive collation:
```sql
create collation case_insensitive (provider = icu, locale = 'und-u-ks-level2', deterministic = false);
```

Current result is an error stating: 
```
ICU is not supported in this build
```

**Comment by kiwicopple** [comment]
thanks @lukaspili - moving this over to the postgres repo

**Comment by xjasonlyu** [comment]
Hi, any progress on this enhancement?

**Comment by steve-chavez** [comment]
Seems postgres wasn't compiled with ICU support, it needs `./configure --with-icu`.

More details at: https://www.2ndquadrant.com/en/blog/icu-support-postgresql-10/

**Comment by BernieSumption** [comment]
ICU is supported on the postgres development server (`supabase start`), so I've just developed and tested a new feature using ICU collations and only discovered that it's not supported in production when I tried to `supabase db push` the new schemas.

So at least the development server is compiled with ICU support. I thought the whole point of the dockerised development server was that you developed against the same binary that you deployed on?
